src/test/java/io/cdap/plugin/sfmc/source/MarketingCloudRecordReaderTest.java

```diff
@@ -0,0 +1,293 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.sfmc.source;
+
+import com.custom.fuelsdk.ETClickEvent;
+import com.custom.fuelsdk.PaginationETSoapObject;
+import com.exacttarget.fuelsdk.ETApiObject;
+import com.exacttarget.fuelsdk.ETClient;
+import com.exacttarget.fuelsdk.ETConfiguration;
+import com.exacttarget.fuelsdk.ETDataExtensionRow;
+import com.exacttarget.fuelsdk.ETResponse;
+import com.exacttarget.fuelsdk.ETResult;
+import com.exacttarget.fuelsdk.ETSoapConnection;
+import com.exacttarget.fuelsdk.ETSoapObject;
+import com.exacttarget.fuelsdk.internal.EventType;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.plugin.PluginProperties;
+import io.cdap.plugin.sfmc.source.util.MarketingCloudColumn;
+import io.cdap.plugin.sfmc.source.util.MarketingCloudObjectInfo;
+import io.cdap.plugin.sfmc.source.util.SourceObject;
+import io.cdap.plugin.sfmc.source.util.SourceQueryMode;
+
+import org.junit.Assert;
+import org.junit.Assume;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.internal.AssumptionViolatedException;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ETClient.class, ClassLoader.class, MarketingCloudClient.class, ETSoapConnection.class})
+public class MarketingCloudRecordReaderTest {
+
+  protected static final String CLIENT_ID = "clientId";
+  protected static final String CLIENT_SECRET = "clientSecret";
+  protected static final String AUTH_ENDPOINT = "authEndPoint";
+  protected static final String SOAP_ENDPOINT = "soapEndPoint";
+  private static final Logger LOG = LoggerFactory.getLogger(MarketingCloudInputFormat.class);
+
+  @Rule
+  public ExpectedException thrown = ExpectedException.none();
+  private MarketingCloudSourceConfig marketingCloudSourceConfig;
+  private MarketingCloudRecordReader marketingCloudRecordReader;
+
+
+  @Before
+  public void initialize() {
+    try {
+      Assume.assumeNotNull(CLIENT_ID, CLIENT_SECRET, AUTH_ENDPOINT, SOAP_ENDPOINT);
```

Comment:
Assume.assumeNotNull() is not needed because you are defining these constants above.

Reply:
Removed